tests/CustomerTest.php

```diff
@@ -137,30 +137,43 @@ public function testCanUpdateSubscription()
         $this->stubRequest(
             'post',
             '/v1/customers/' . $customer->id . '/subscription',
-            array("plan" => "plan")
+            array("plan" => "plan"),
+            array(
+                "object" => "subscription",
+                "id" => "sub_foo"
+            )
         );
-        $customer->updateSubscription(array("plan" => "plan"));
+        $resource = $customer->updateSubscription(array("plan" => "plan"));
+        $this->assertSame("Stripe\\Subscription", get_class($resource));
+        $this->assertSame("sub_foo", $customer->subscription->id);
     }
 
     public function testCanCancelSubscription()
     {
         $customer = Customer::retrieve(TEST_RESOURCE_ID);
         $this->stubRequest(
             'delete',
-            '/v1/customers/' . $customer->id . '/subscription'
+            '/v1/customers/' . $customer->id . '/subscription',
+            array(),
+            array(
+                "object" => "subscription",
+                "id" => "sub_foo"
+            )
         );
-        $customer->cancelSubscription();
+        $resource = $customer->cancelSubscription();
+        $this->assertSame("Stripe\\Subscription", get_class($resource));
+        $this->assertSame("sub_foo", $customer->subscription->id);
     }
 
     public function testCanDeleteDiscount()
     {
         $customer = Customer::retrieve(TEST_RESOURCE_ID);
         $this->stubRequest(
             'delete',
-            '/v1/customers/' . $customer->id . '/discount'
+            '/v1/customers/' . $customer->id . '/discount',
         );
-        $resource = $customer->deleteDiscount();
-        $this->assertSame($resource, null);
+        $customer->deleteDiscount();
+        $this->assertSame($customer->discount, null);
```

Comment:
While this works, this is IMO incorrect. stripe-mock does not handle discount in the first place and it's null mostly by luck no?

Reply:
nevermind this is stubbed hence why it did not do what I expected it to do

Reply:
nevermind this is stubbed hence why it did not do what I expected it to do